src/OpenApi/src/Helpers/OpenApiTagComparer.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+/// <summary>
+/// This comparer is used to maintain a globally unique list of tags encountered
+/// in a particular OpenAPI document.
+/// </summary>
+internal class OpenApiTagComparer : IEqualityComparer<OpenApiTag>
+{
+    public static OpenApiTagComparer Instance { get; } = new OpenApiTagComparer();
+
+    public bool Equals(OpenApiTag? x, OpenApiTag? y)
+    {
+        if (x is null && y is null)
+        {
+            return true;
+        }
+        if (x is null || y is null)
+        {
+            return false;
+        }
+        return x.Name == y.Name;
```

Comment:
Should this be case sensitive or insensitive? I wonder if using `string.Equals()` with the appropriate ordinal comparison would make the intent clearer in either case?

Reply:
Good question! The OpenAPI specification states that all field _names_ should be considered case sensitive ([ref](https://spec.openapis.org/oas/v3.1.0#format)), but it doesn't provide an opinion on field _values_.  For consistencies sake, I think it makes sense to apply the same sensitivity rules to values as well.

I'll pop a comment in this code clarifying why this case sensitive comparison is used.